File: prog-beans-registration/src/main/java/com/jetbrains/demo/InMemoryCacheManager.java
package com.jetbrains.demo;

import java.util.HashMap;
import java.util.Map;

public class InMemoryCacheManager implements CacheManager {
    private final Map<String, Object> cache = new HashMap<>();

    @Override
    public void put(String key, Object value) {
        cache.put(key, value);
    }

    @Override
    public Object get(String key) {
        return cache.get(key);
    }

    @Override
    public void clearCache() {
        cache.clear();
    }
}

File: kotlin-bean-registration/src/main/kotlin/com/jetbrains/demo/InMemoryCacheManager.kt
package com.jetbrains.demo


class InMemoryCacheManager : CacheManager {
    private val cache: MutableMap<String, Any> = HashMap<String, Any>()

    override fun put(key: String, value: Any) {
        cache[key] = value
    }

    override fun get(key: String): Any? {
        return cache[key]
    }

    override fun clearCache() {
        cache.clear()
    }
}